ci/infra/testrunner/tests/conftest.py

```diff
@@ -81,3 +81,60 @@ def kubectl(conf, target):
 def platform(conf, target):
     platform = platforms.get_platform(conf, target)
     return platform
+
+
+@pytest.fixture
+def cluster_ready(kubectl):
+    def func():
+        wait(kubectl.run_kubectl, 'wait --for=condition=Ready node --all --timeout=1m', wait_delay=60, wait_timeout=300, wait_backoff=30, wait_retries=5)
+        wait(kubectl.run_kubectl, 'wait --timeout=1m --for=condition=Ready pods --all --namespace=kube-system --field-selector=status.phase!=Succeeded', wait_delay=60, wait_timeout=300, wait_backoff=30, wait_retries=5, wait_allow=(RuntimeError))
+
+    return func
+
+
+@pytest.fixture
+def provision_master(request, conf, platform):
+    def cleanup():
+        if platform.get_num_nodes('master') > conf.master.count:
```

Comment:
Here is a problem. The number of masters and workers can be override with command line parameters in testrunner. If we provision the cluster and then execute a tests that uses this fixture (with the skip-setup option), the number of nodes that are defined in the conf doesn't match the number actually provisioned.  I would do something like this:
```
dec_provision_master(request, conf, platform):
    master = platform.get_num_nodes('master')
    def cleanup():
          platform.remove_nodes('master', master)
```
Then the 'remove_nodes'  function should check the node exists before removing it.

Reply:
removed